erpnext/manufacturing/doctype/production_plan/production_plan.json

```diff
@@ -358,7 +358,7 @@
    "options": "Production Plan Sub Assembly Item"
   },
   {
-   "depends_on": "eval:doc.po_items && doc.po_items.length && doc.docstatus == 0",
+   "depends_on": "eval:doc.po_items && doc.po_items[0].item_code && doc.docstatus == 0",
```

Comment:
why item code not added by the User in po_items table?

Reply:
Add condition for sub_assembly_items

Reply:
> why item code not added by the User in po_items table?

By default, we are provided with an empty row (item_code not selected) in **po_items** while creating a new Production Plan. The current condition applied on **Get Sub Assembly Items** button is **po_items && po_items.length**, which is always true when we create a new Production Plan, which means the **Get Sub Assembly Items** is visible. If there is no **po_item(s)** and the user tries to get sub-assembly items by clicking on **Get Sub Assembly Items**, an error will be thrown. Ideally, the **Get Sub Assembly Items** should be hidden if there are no **po_items**.

We have different ways to do the same:
1. Check the tem_code in the 0th index. (This PR)
2. Checking each row in JS and hiding the button accordingly.

Reply:
> Add condition for sub_assembly_items

Want to hide the **Get Sub Assembly Items** button, based on **po_items**.

Reply:
Description Image: In this case, if we click on "Get Sub Assembly Items" an error will be thrown saying that **'NoneType' object is not iterable"**